ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/ccd/repositories/CCDCaseApi.java

```diff
@@ -0,0 +1,109 @@
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@ConditionalOnProperty(prefix = "core_case_data", name = "api.url")
+public class CCDCaseApi {
+
+    public static final String JURISDICTION_ID = "CMC";
+    public static final String CASE_TYPE_ID = "MoneyClaimCase";
+
+    private final CoreCaseDataApi coreCaseDataApi;
+    private final AuthTokenGenerator authTokenGenerator;
+    private final UserService userService;
+    private final CaseMapper caseMapper;
+    private final JsonMapper jsonMapper;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CCDCaseApi.class);
+
+    public CCDCaseApi(
+        CoreCaseDataApi coreCaseDataApi,
+        AuthTokenGenerator authTokenGenerator,
+        UserService userService,
+        CaseMapper caseMapper,
+        JsonMapper jsonMapper
+    ) {
+        this.coreCaseDataApi = coreCaseDataApi;
+        this.authTokenGenerator = authTokenGenerator;
+        this.userService = userService;
+        this.caseMapper = caseMapper;
+        this.jsonMapper = jsonMapper;
+    }
+
+    public Optional<Map<String, Object>> getByReferenceNumber(String authorisation, String referenceNumber) {
+        LOGGER.info("Get claim from CCD " + referenceNumber);
+        User user = userService.getUser(authorisation);
+
+        List<Map<String, Object>> claims = extractClaims(
+            search(user, ImmutableMap.of("case.referenceNumber", referenceNumber))
+        );
+
+        if (claims.size() > 1) {
+            throw new RuntimeException("More than one claim found by claim reference " + referenceNumber);
+        }
+
+        LOGGER.info("Claim found " + claims.size());
+
+        return claims.isEmpty() ? Optional.empty() : Optional.of(claims.get(0));
+    }
+
+    private List<CaseDetails> search(User user, Map<String, Object> searchString) {
+
+        String serviceAuthToken = this.authTokenGenerator.generate();
+
+        List<CaseDetails> result;
+        result = this.coreCaseDataApi.searchForCaseworker(
+            user.getAuthorisation(),
+            serviceAuthToken,
+            user.getUserDetails().getId(),
+            JURISDICTION_ID,
+            CASE_TYPE_ID,
+            searchString
+        );
+
+        return result;
+    }
+
+    private List<Map<String, Object>> extractClaims(List<CaseDetails> result) {
+
+        Map<Long, Map<String, Object>> collectMap = result.stream()
+            .collect(Collectors.toMap(CaseDetails::getId, CaseDetails::getData));
+
+        return collectMap.entrySet().stream()
+            .map(this::mapToClaim)
+            .collect(Collectors.toList());
+    }
+
+    private CCDCase convertToCCDCase(Map<String, Object> mapData) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused private "convertToCCDCase" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AUnusedPrivateMethod)